config/checkstyle-examples-checks.xml

```diff
@@ -0,0 +1,82 @@
+<?xml version="1.0"?>
+<!DOCTYPE module PUBLIC
+        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
+        "https://checkstyle.org/dtds/configuration_1_3.dtd">
+
+<module name="Checker">
+  <property name="charset" value="UTF-8"/>
+  <property name="haltOnException" value="false"/>
+
+  <property name="cacheFile" value="target/cache_xdocs_examples"/>
+
+  <module name="SuppressionFilter">
+    <property name="file" value="${checkstyle.examples-suppressions.file}"/>
+  </module>
+
+  <module name="FileTabCharacter">
+    <property name="eachLine" value="false"/>
+  </module>
+
+  <module name="FileLength">
+    <property name="max" value="45"/>
+  </module>
+
+  <module name="RegexpOnFilename">
+    <property name="id" value="exampleFileName"/>
+    <property name="fileNamePattern" value="^Example\d+"/>
```

Comment:
Why not `^Example\d+\.`?
If so, violation message should be updated too.

Reply:
Changed to `^Example\d+\$`. We are ignoring the file extension.

Reply:
@rnveach 
Would that be okay?
```xml
  <module name="RegexpOnFilename">
    <property name="id" value="exampleFileName"/>
    <property name="fileNamePattern" value="^Example\d+$"/>
    <property name="ignoreFileNameExtensions" value="true"/>
    <property name="match" value="false"/>
    <message key="regexp.filename.mismatch"
             value="All example files should only match '^Example\d+$' exactly"/>
  </module>
```

Reply:
It is fine, but I still think full regexp here is a little overkill. Also if you are going to do full regexp, I think you can just use `{1}` as they are applied to the message by default somehow.
https://github.com/search?q=path%3Asrc%2Fmain%2Fresources%2Fcom%2Fpuppycrawl%2Ftools%2Fcheckstyle%2Fchecks%2Fregexp+path%3A**%2Fmessages*.properties+repo%3Acheckstyle%2Fcheckstyle+%22regexp.filename.mismatch%22&type=code

Reply:
Yes, it works.
```diff
$ git diff
─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
modified: config/checkstyle-examples-checks.xml
─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
@ config/checkstyle-examples-checks.xml:30 @
    <property name="ignoreFileNameExtensions" value="true"/>
    <property name="match" value="false"/>
    <message key="regexp.filename.mismatch"
-             value="All example files should only match '^Example\d+$' exactly"/>
+             value="All example files should only match ''{1}'' exactly"/>
  </module>

  <module name="RegexpSingleline">
_____________________________________________________________________________________________________________________________________________________________________________
 stoyan @  zenbook:  ~/open-source/checkstyle                                                                                                          issue/13116 [!?↓↑]
$ git status
On branch issue/13116
Your branch and 'origin/issue/13116' have diverged,
and have 6 and 1 different commits each, respectively.
  (use "git pull" to merge the remote branch into yours)

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   config/checkstyle-examples-checks.xml

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/asd.java

no changes added to commit (use "git add" and/or "git commit -a")

```
```
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abbreviationaswordinname/asd.java:1: All example files should only match '^Example\d+$' exactly [exampleFileName]
```

Reply:
Done.